components/select/Select.razor.cs

```diff
@@ -637,7 +643,7 @@ private void CreateDeleteSelectOptions()
 
             foreach (var item in _datasource)
             {
-                TItemValue value = _getValue(item);
+                TItemValue value = _getValue == null ? THelper.ChangeType<TItemValue>(item) : _getValue(item);
```

Comment:
@Zonciu Could you look at this? Could this be incorporated somehow into expression tree idea? Maybe it is not worth it, or too complex to actually invest time in it.

Reply:
I also thought about this. I think it's nice to get `TItem x => (TItemValue)x` for string.Empty and whitespace. But maybe it's better to return null or throw an exception for null.